File: algorithms/2_medium/S/01291_sequential_digits/solution.cpp
constexpr int max_hi = 1'000'000'000;

class Solution {
public:
    static vector<int> sequentialDigits(const int lo, const int hi) {
        assert(lo >= 10 && lo <= hi && hi <= max_hi);
        auto res = vector<int>();

        for (int start_dig = 1; start_dig <= 9; start_dig += 1) {
            int num = 0;

            for (int dig = start_dig; dig <= 9; dig += 1) {
                num = num * 10 + dig;

                if (num > hi) {
                    break;
                }

                if (num >= lo) {
                    res.push_back(num);
                }
            }
        }

        ranges::sort(res);
        return res;
    }
};

File: algorithms/2_medium/S/01291_sequential_digits/solution.rs
impl Solution {
    const MAX_HI: i32 = 1_000_000_000;

    pub fn sequential_digits(lo: i32, hi: i32) -> Vec<i32> {
        assert!(lo >= 10 && lo <= hi && hi <= Self::MAX_HI);
        let mut res: Vec<i32> = Vec::new();

        for start_dig in 1..=9_i32 {
            let mut num = 0_i32;

            for dig in start_dig..=9_i32 {
                num = num * 10 + dig;

                if num > hi {
                    break;
                }

                if num >= lo {
                    res.push(num);
                }
            } 
        }

        res.sort_unstable();
        res
    }
}
